### PR TITLE
[ROCKETMQ-254] fix logger appender unit test too long with mock

### PR DESCRIPTION
**Pull request by lindzh**
LoggerAppender unit test cost much time, and in some cases it may cause test failure,fix this case this mockito.

### CODE REVIEW

File: logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java [hunk 1]
```diff
@@ -61,7 +68,7 @@ public static MQProducer getInstance(String nameServerAddress, String group) thr
         defaultMQProducer.setNamesrvAddr(nameServerAddress);
         MQProducer beforeProducer = null;
         //cas put producer
```
Comment (vongosling) [inline logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java]: what's the purpose for this comment :-)

Comment (lindzh) [inline logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java]: This comment has been deleted.

**Comment by vongosling** [comment]
LGTM

File: logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java [hunk 2]
```diff
@@ -70,24 +76,28 @@ public static MQProducer getInstance(String nameServerAddress, String group) thr
...
+    public void closeAll() {
+        Set<Map.Entry<String, MQProducer>> entries = getProducerInstance().producerMap.entrySet();
         for (Map.Entry<String, MQProducer> entry : entries) {
-            producerMap.remove(entry.getKey());
+            getProducerInstance().producerMap.remove(entry.getKey());
             entry.getValue().shutdown();
         }
     }
 
+    public static void setInstance(ProducerInstance instance1) {
```
Comment (shroman) [inline logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java]: looks pretty dangerous to me.
Especially considering you make it only for tests.

Comment (lindzh) [inline logappender/src/main/java/org/apache/rocketmq/logappender/common/ProducerInstance.java]: Yes ,it is dangerous for only test and this problem has been fixed with reflect.

**Comment by lollipopjin** [comment]
Thanks @lindzh. I have merged it, please close this pr :-)

**Pull request closed without merging**